tools/invoke-tests

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash -e
+
+source_directory=$(dirname "$0")/..
+source_directory=$(realpath "$source_directory")
+
+usage() {
+    echo "usage: $0 [options] [test-file...]
+options:
+    --coverage         record test coverage
+    --prove-tool       prove tool to use (defaults to prove)
+    --make-tool        make tool to use (defaults to make)
+    --build-directory  build directory (defaults to $source_directory)"
+    exit
+}
+
+# parse arguments
+prove_path=prove make_path=make build_directory=$source_directory
+opts=$(getopt --options h --longoptions help,coverage,prove-tool:,make-tool:,build-directory: --name "$(basename "$0")" -- "$@") || usage
+eval set -- "$opts"
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    -h | --help       ) usage;                shift   ;;
+    --coverage        ) WITH_COVER_OPTIONS=1; shift   ;;
+    --prove-tool      ) prove_path=$2;        shift 2 ;;
+    --make-tool       ) make_path=$2;         shift 2 ;;
+    --build-directory ) build_directory=$2;   shift 2 ;;
+    --                ) shift; break ;;
+    *                 ) break ;;
+  esac
+done
+test_files=("$@")
+
+# set Perl module include path and coverage options
+export PERL5LIB="$source_directory:$source_directory/ppmclibs:$source_directory/ppmclibs/blib/arch/auto/tinycv:$PERL5LIB"
+if [[ $WITH_COVER_OPTIONS ]]; then
+    db="$build_directory/cover_db"
+    path_regex="^|$source_directory/|\\.\\./"
+    select="($path_regex)(OpenQA|backend|consoles|ppmclibs)/|($path_regex)isotovideo|($path_regex)[^/]+\.pm,-ignore,\.t|data/tests/|fake/tests/|$prove_path"
+    export PERL5OPT="-MDevel::Cover=-db,$db,-select,$select,-coverage,statement"
+fi
+
+# set variables for tests which need to invoke the make tool
+export OS_AUTOINST_BUILD_DIRECTORY=$build_directory
+export OS_AUTOINST_MAKE_TOOL=$make_path
+
+# allow specifying the list of tests via TESTS in consistency with openQA; otherwise run all tests
+[[ ${#test_files[@]} -lt 1 ]] && test_files=($TESTS)
+[[ ${#test_files[@]} -lt 1 ]] && test_files=(-r)
+
+# invoke tests within the t directory
+cd t && "$prove_path" $PROVE_ARGS "${test_files[@]}"
```

Comment:
IMHO that's overly complicated and we should be able to use pure POSIX-sh syntax and just define:

```shell
TESTS="$@"
TESTS="${TESTS:-"-r"}"
cd t && "$prove_path" $PROVE_ARGS $TESTS
```

wouldn't that work?

Reply:
It would work as long as there's no space within the path of a test file. Of course with my code it also only works for test files specified as CLI argument so I guess we can remove that and just never use spaces.